airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -460,6 +465,47 @@ class ContactListStream(Stream):
     limit_field = "count"
 
 
+class DealStageHistoryStream(Stream):
+    """Deal stage history, API v1
+    Deal stage history is exposed by the v1 API, but not the v3 API.
+    The v1 endpoint requires the contacts scope.
+    Docs: https://legacydocs.hubspot.com/docs/methods/deals/get-all-deals
+    """
+    
+    url = "/deals/v1/deal/paged"
+    more_key = "hasMore"
+    data_field = "deals"
+    
+    def parse_cursor(self, record):
+        updated_at = 0
+        for prop in record.get("properties", {}).values():
+            if prop.get("timestamp", 0) > updated_at:
+                updated_at = prop.get("timestamp")
+        return updated_at
+    
+    def list(self, fields) -> Iterable:
+        params = {"propertiesWithHistory": "dealstage"}
+        yield from self.read(partial(self._api.get, url=self.url), params)
+
+
+class DealStream(CRMObjectStream):
+    """Deals, API v3"""
+    
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.stageHistory = DealStageHistoryStream(**kwargs)
+    
+    def list(self, fields) -> Iterable:
```

Comment:
unfortunately this doesn't work as intended, dealstage will never be retrieved because it is located in `deals->properties->dealstage`

Reply:
Applied your suggestions here https://github.com/airbytehq/airbyte/pull/3734/files#r644783984